File: 151-reverse-words-in-a-string/main.py
class Solution:
    def reverseWords(self, s: str) -> str:
        
        l = s.split()
        n = len(l)
        for i in range(n// 2):
            l[i], l[n - i - 1] = l[n - i - 1], l[i]

        return " ".join(l)

        

File: 151-reverse-words-in-a-string/main.rs
impl Solution {
    pub fn reverse_words(s: String) -> String {
        
        let mut l: Vec<&str> = s.split_whitespace().collect();
        let n = l.len();

        for i in 0..(n / 2) {
            l.swap(i, n - i - 1);
        }

        l.join(" ").to_string()
    }
}
